### ISSUE TITLE
[Bug]: Crash when running scripts in isolation from command line

### ISSUE THREAD

**Issue by d0n3val** [DmitriySalnikov/godot_debug_draw_3d#59]
### Godot version

4.3

### DebugDraw3D version

1.4.5

### On which operating systems the error occurs

Windows, Linux

### Issue description

After installing the addon everything works well, but when that same project uses the command line of godot to execute a script, the is a crash inside debug draw library

windows:
![image](https://github.com/user-attachments/assets/3f925b31-219d-408f-bdfb-6ca101721af1)

Linux:
![image](https://github.com/user-attachments/assets/1a9ba584-8bfd-41ad-baf4-1e12b31e06ef)

### Steps to reproduce

Create an simple project with latest godot stable 4.3 and latest release debug draw libraries. 
Use the godot command line to like "godot --headless --check-only -path PROJECT --script SOMETHING.gd

Godot crashes

Note: I've tried removing the "--headless" flag and the crash still happens

### Minimal reproduction project

Repro steps are trivial, it happens both in linux and windows with any script you try to compile.

**Comment by DmitriySalnikov** [comment]
It's strange, because every commit in CI verifies that godot is running in headless mode. But only with Godot version 4.2.

---

Well, I found out under what conditions the crash occurs.

This command is working well:
```bash
> godot.windows.editor.dev.x86_64.mono.exe  -v --headless --path . --script res://headless_test.gd

Godot Engine v4.3.stable.mono.custom_build.77dcf97d8 (2024-08-14 23:00:16 UTC) - https://godotengine.org
TextServer: Added interface "Dummy"
TextServer: Added interface "ICU / HarfBuzz / Graphite (Built-in)"
Using "default" pen tablet driver...

TextServer: Primary interface set to: "ICU / HarfBuzz / Graphite (Built-in)".
.NET: Initializing module...
Found hostfxr: C:\Program Files\dotnet\host/fxr/8.0.10/hostfxr.dll
.NET: hostfxr initialized
.NET: GodotPlugins initialized
CORE API HASH: 4070621931
EDITOR API HASH: 591096730

Start of testing.
frustum_length_scale: 0.07000000029802
End of testing.
Loaded system CA certificates
Unloading: Disposing tracked instances...
Unloading: Finished disposing tracked instances.
XR: Clearing primary interface
XR: Removed interface "Native mobile"
XR: Removed interface "OpenXR"
```

With this script:
```gdscript
extends SceneTree


func _init():
	DebugDrawManager.debug_enabled = true
	print()
	print("Start of testing.")
	
	if true:
		var _s = DebugDraw3D.new_scoped_config().set_thickness(0.1)
		DebugDraw3D.draw_box(Vector3.ZERO, Quaternion.IDENTITY, Vector3.ONE, DebugDraw3D.empty_color, true, 1.2)
	DebugDraw3D.draw_sphere(Vector3.ZERO, 0.5, DebugDraw3D.empty_color, 1.5)
	DebugDraw3D.draw_gizmo(Transform3D().scaled_local(Vector3.ONE*0.4), Color.BROWN, true, 2.5)
	DebugDraw2D.set_text("FPS", "0")
	DebugDraw2D.begin_text_group("Info")
	DebugDraw2D.set_text("INFO", "hello", 1, DebugDraw2D.empty_color, 1)
	DebugDraw2D.end_text_group()
	
	DebugDraw3D.draw_sphere(Vector3.ZERO)
	DebugDraw3D.config.frustum_length_scale = 0.07
	print("frustum_length_scale: ", DebugDraw3D.config.frustum_length_scale)
	
	DebugDrawManager.clear_all()
	
	print("End of testing.")
	quit()
```

But this command causes a crash:
```bash
godot.windows.editor.dev.x86_64.mono.exe  -v --headless --check-only --path . --script res://headless_test.gd
```

That is, the problem is somewhere in the `--check-only`.

**Comment by DmitriySalnikov** [comment]
Fixed in 742443c26f1de3743262f3839c79650a37236cdb

Can you test this [build](https://github.com/DmitriySalnikov/godot_debug_draw_3d/actions/runs/11825868857)?

**Comment by d0n3val** [comment]
> Fixed in [742443c](https://github.com/DmitriySalnikov/godot_debug_draw_3d/commit/742443c26f1de3743262f3839c79650a37236cdb)
> 
> Can you test this [build](https://github.com/DmitriySalnikov/godot_debug_draw_3d/actions/runs/11825868857)?

Compiled latest code in linux and works perfectly, cannot test on windows right now.

Thanks so much for answering so fast!
